Helpers/Eligibility.php

```diff
@@ -456,32 +466,38 @@ public function hasEnabledPaymentPlansInBo():bool
     {
         $hasActivePlans = false;
         $inConfigPaymentPlans = $this->getEnabledConfigPaymentPlans();
-        $this->logger->info('In hasEnabledPaymentPlansInBo',[$inConfigPaymentPlans]);
         foreach ($inConfigPaymentPlans as $paymentPlan) {
             if($paymentPlan->isEnabled()){
-                $this->logger->info('Has an active plan',[$paymentPlan]);
-                $hasActivePlans = true;
+                return true;
             }
         }
         return $hasActivePlans;
     }
 
-    private function buildType($installmentCount,$isDeferred):string
+    /**
+     * @param $eligibilities [$paymentPlan]
+     * @return array
+     */
+    public function sortEligibilities($eligibilities):array
     {
-        $type = 'other';
+        $sortedEligibilities=[];
+        foreach ($eligibilities as $paymentPlan){
+
+            $planConfig = $paymentPlan->getPlanConfig();
+            $planKey = $planConfig->PlanKey();
+
+            $type = $this->getPaymentType($planKey);
+            $sortedEligibilities[$type][]=$paymentPlan;
 
-        if ($installmentCount > 1 && !$isDeferred) {
-            $type = self::INSTALLMENTS_TYPE;
-        }
-        if ($installmentCount > 4 && !$isDeferred) {
-            $type = self::SPREAD_TYPE;
-        }
-        if ($installmentCount == 1 && $isDeferred) {
-            $type = self::DEFFERED_TYPE;
         }
-        return $type;
+        return $sortedEligibilities;
     }
 
+    /**
+     * Get payment type according to plan key ( like general:10:0:0 ) for split ordering
+     * @param $planKey
```

Comment:
add type of param (in the docblock and also in the method signature if possible)